pkg/env/helpers.go

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package env
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/fatih/structs"
+)
+
+type descriptionTree struct {
+	Value    string
+	Children []*descriptionTree
+}
+
+func newDescriptionTree(root string) *descriptionTree {
+	return &descriptionTree{
+		Value:    root,
+		Children: nil,
+	}
+}
+
+func (t *descriptionTree) AddNode(tree *descriptionTree) {
+	if t.Children == nil {
+		t.Children = []*descriptionTree{tree}
+		return
+	}
+	t.Children = append(t.Children, tree)
+}
+
+type configurationParameter struct {
+	Name         string
+	DefaultValue interface{}
+}
+
```

Comment:
Does it make sense to refactor the next two functions like so:

```go
func buildParametersAndDescriptions(value interface{}) ([]configurationParameter, []string) {
	tree := &descriptionTree{}
	return buildParameters(tree, value), buildDescriptionsFromTree(tree)
}

func buildParameters(tree *descriptionTree, value interface{}) []configurationParameter {
	var parameters []configurationParameter
	buildDescriptionTreeWithParameters(value, tree, "", &parameters)
	return parameters
}
```

This way we can eliminate some code duplication. Then all clients can use the **buildParametersAndDescriptions** and if they don't need the descriptions, they can simply ignore the second variable - like here: https://github.com/Peripli/service-manager/blob/067e7c84d83885e084d04be62980f18de31ea53a/pkg/env/env.go#L114